docs/ref/forms/fields.txt

```diff
@@ -913,6 +913,12 @@ For each field, we describe the default widget used if you don't specify
     * Normalizes to: A Python ``True``, ``False`` or ``None`` value.
     * Validates nothing (i.e., it never raises a ``ValidationError``).
 
+    .. tip::
+        Supply ``choices=((True, 'Custom True'), (False, 'Custom False'))``
+        to :class:`~django.db.models.BooleanField` in order to render a
+        :class:`~django.forms.Select` or :class:`~django.forms.RadioSelect`
+        widget on the resulting :class:`TypedChoiceField`.
+
```

Comment:
Can't we just provide the example of using a `Select` passing `choices` (which I think was [the suggestion on the ticket](https://code.djangoproject.com/ticket/23681#comment:7).) 

```suggestion
``NullBooleanField`` may be used with e.g. ``Select`` or ``RadioSelect``
widgets providing custom ``choices``::

    NullBooleanField(
        widget=Select(
            choices=[
                ('', 'Unknown'),
                (True, 'Yes'),
                (False, 'No'),
            ]
        )
    )
```

I'm think this kind of thing would close this and ticket-27445 that @smithdc1 is working on in #12946 
Thoughts?

Reply:
Yeah, I think this will do. It helps to know this is possible without additional development, so thanks for reviewing #12946. Will adopt your suggestion and push again.

Reply:
Hi both.

I'm a bit swamped at the moment with life generally. I'd like to spend some time reminding myself of all of this but suspect that may be a few days before I get some proper thinking time. Initial reaction is that this was a bit 'fiddley' and as Carlton mentioned my patch mixes issues. Given that I suspect the suggested doc fix is reasonabe. 

Part of me wonders what would happen if someone posted this question on stackoverflow. I'd have thought folk would find a way to make it work, so we should go with that (being a doc change) rather than making the code more complex?